.github/workflows/format-code.yml

```diff
@@ -58,21 +58,22 @@ jobs:
       - name: Check for changes
         run: |
           git add .
-          git commit -m "Updates from GitHub Actions Format Code workflow" || true
           branch_name=$(git branch --show-current)
-          changes=$(git diff origin/main...$branch_name --name-only)
-          if [ -z "$changes" ]; then
+          if git diff --cached --quiet; then
             echo "No changes detected."
           else
             echo "Changes detected."
             echo "changes=true" >> $GITHUB_ENV
           fi
-
-      - name: Push changes
-        if: env.changes == 'true'
-        run: |
-          git config --global push.autoSetupRemote true
-          git push
+  
+      - name: Generate signed commit for any changes that may exist
+        uses: planetscale/ghcommit-action@c7915d6c18d5ce4eb42b0eff3f10a29fe0766e4c #v1.44
```

Comment:
We should be careful when using external actions from 3rd parties. Is it too difficult to achieve the same without using a GH action? Have you done some research on the action provider?

Reply:
The above provider is one that was recommended by @jacobwoffenden in Analytics Platforms and is used by them for the same purpose. Without this, the other options would be to 
- Store & maintain our own GPG keys (which is an administrative burden for all of our repositories),
- Use the GraphQL API which is significantly more work involved. 
- The other option would be to temporarily disable signed commits for the workflow but I'm not in favour of that.

In short there is no simple or straight forward solution that I have been able to determine through this ticket.